File: LeetCode-Problems/Easy/Matrix-Diagonal-Sum.java
//Java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0;
        
        int j = mat.length-1;
        for (int i = 0; i < mat.length; i++){
            int addLR = mat[i][i];
            int addRL = mat[i][j];
            sum = sum + addLR + addRL;
            j--;
        }
        if (mat.length % 2 == 1){
            int middle = mat[mat.length/2][mat.length/2];
            sum = sum - middle;
        }
        return sum;
    }
}

File: LeetCode-Problems/Easy/Matrix-Diagonal-Sum.py
# Python
class Solution(object):
    def diagonalSum(self, mat):
        """
        :type mat: List[List[int]]
        :rtype: int
        """
        rSum = 0
        y = len(mat)-1
        for x in range (0, len(mat)):
            addLR = mat[x][x]
            addRL = mat[x][y]
            rSum = rSum + addLR + addRL
            y = y-1
        if (len(mat) % 2 == 1):
            rSum = rSum - mat[len(mat)/2][len(mat)/2]
        return rSum
        
            
            
